homeassistant/components/vicare/binary_sensor.py

```diff
@@ -203,7 +214,7 @@ def device_info(self) -> DeviceInfo:
         """Return device info for this device."""
         return DeviceInfo(
             identifiers={(DOMAIN, self._device_config.getConfig().serial)},
-            name=self._device_config.getModel(),
+            name=f"{self._device_config.getModel()}-{self._device_config.getConfig().serial}",
```

Comment:
Looks like this is actually the serial of the gateway and not of the device. So a gateway could provide multiple devices and the serial of device is not exposed.

https://github.com/somm15/PyViCare/blob/8ba411483a865e074d1146fd1b8b7a8c4f4be122/PyViCare/PyViCare.py#L72-L73

Reply:
This will get us in trouble once we update to PyViCare ~2.28.x where the gateways will be exposed.

Maybe we should get the right serial here, instead of the gateways one.

<img width="1040" alt="Bildschirmfoto 2023-10-13 um 23 25 19" src="https://github.com/home-assistant/core/assets/9592452/177b09c0-289d-4a7b-93b9-2cfc9034dd9f">